BridgeAppSDK/SBAKeychainWrapper.m

```diff
@@ -0,0 +1,317 @@
+                                                 code:status
+                                             userInfo:@{NSLocalizedDescriptionKey: [Localization localizedString:@"KEYCHAIN_ADD_ERROR_MESSAGE"]}];
+                }
+                returnValue = NO;
+            }
+        } else {
+            returnValue = NO;
+        }
+    }
+    return returnValue;
+}
+
+- (BOOL)removeItemForKey:(NSString *)key
+                 service:(NSString *)service
+             accessGroup:(NSString *)accessGroup
+                   error:(NSError **)error {
+    BOOL returnValue = NO;
+    if (key) {
+        if (!service) {
+            service = SBAKeychainWrapperDefaultService();
+        }
+        
+        NSMutableDictionary *itemToDelete = [[NSMutableDictionary alloc] init];
+        [itemToDelete setObject:(__bridge id)kSecClassGenericPassword forKey:(__bridge id)kSecClass];
+        [itemToDelete setObject:service forKey:(__bridge id)kSecAttrService];
+        [itemToDelete setObject:key forKey:(__bridge id)kSecAttrAccount];
+#if !TARGET_IPHONE_SIMULATOR && defined(__IPHONE_OS_VERSION_MIN_REQUIRED)
+        if (accessGroup) {
+            [itemToDelete setObject:accessGroup forKey:(__bridge id)kSecAttrAccessGroup];
+        }
+#endif
+        
+        OSStatus status = SecItemDelete((__bridge CFDictionaryRef)itemToDelete);
+        if (status != errSecSuccess && status != errSecItemNotFound) {
+            if (error) {
+                *error = [NSError errorWithDomain:NSOSStatusErrorDomain
+                                             code:status
+                                         userInfo:@{NSLocalizedDescriptionKey: [Localization localizedString:@"KEYCHAIN_DELETE_ERROR_MESSAGE"]}];
+            }
+            returnValue = NO;
+        } else {
+            returnValue = YES;
+        }
+    }
+    return returnValue;
+}
+
+- (BOOL)removeAllItemsForService:(NSString *)service
+                     accessGroup:(NSString *)accessGroup
+                           error:(NSError **)error {
+    NSArray *items = [self itemsForService:service accessGroup:accessGroup error:error];
+    BOOL returnValue = NO;
+    for (NSDictionary *item in items) {
+        NSMutableDictionary *itemToDelete = [[NSMutableDictionary alloc] initWithDictionary:item];
+        [itemToDelete setObject:(__bridge id)kSecClassGenericPassword forKey:(__bridge id)kSecClass];
+        
+        OSStatus status = SecItemDelete((__bridge CFDictionaryRef)itemToDelete);
+        if (status != errSecSuccess) {
+            if (error) {
+                *error = [NSError errorWithDomain:NSOSStatusErrorDomain
+                                             code:status
+                                         userInfo:@{NSLocalizedDescriptionKey: [Localization localizedString:@"KEYCHAIN_DELETE_ERROR_MESSAGE"]}];
+            }
+            returnValue = NO;
+        } else {
+            returnValue = YES;
```

Comment:
Once again, presumably from RK like this, but won't this just end up returning YES or NO based on the status from the last item in the list, regardless of the success or failure of removal of all the previous items? And is this what we want?